packages/backend/src/open_payments/wallet_address/service.ts

```diff
@@ -165,27 +184,50 @@ async function createWalletAddress(
 
 async function updateWalletAddress(
   deps: ServiceDependencies,
-  { id, status, publicName }: UpdateOptions
+  { id, status, publicName, additionalProperties }: UpdateOptions
 ): Promise<WalletAddress | WalletAddressError> {
+  const trx = await WalletAddress.startTransaction()
   try {
     const update: UpdateInput = { publicName }
-    const walletAddress = await WalletAddress.query(deps.knex)
+    const walletAddress = await WalletAddress.query(trx)
       .findById(id)
       .throwIfNotFound()
 
     if (status === 'INACTIVE' && walletAddress.isActive) {
       update.deactivatedAt = new Date()
-      await deactivateOpenIncomingPaymentsByWalletAddress(deps, id)
+      await deactivateOpenIncomingPaymentsByWalletAddress(deps, id, trx)
     } else if (status === 'ACTIVE' && !walletAddress.isActive) {
       update.deactivatedAt = null
     }
 
-    return await walletAddress
-      .$query(deps.knex)
+    const updatedWalletAddress = await walletAddress
+      .$query(trx)
       .patchAndFetch(update)
       .withGraphFetched('asset')
       .throwIfNotFound()
+
+    // Override all existing additional properties if new ones are provided
+    if (additionalProperties) {
```

Comment:
we should add some tests for these

Reply:
added: [770c0bd](https://github.com/interledger/rafiki/pull/2769/commits/770c0bd99d3f24a94dce38a21977a44ae4221478)